### PR TITLE
Create PR summary comment

### LINKED ISSUE #17: Create PR summary
Remove dependency on a third party Action to create the PR summary.

### PR DESCRIPTION
**Pull request by ahtrotta**
Fixes #17 

This PR makes it so that we do not have to use [this GitHub action](https://github.com/thollander/actions-comment-pull-request/tree/main) to create PR comments.

This PR creates a PR summary from the `report.md` file. It will overwrite a previously existing report summary if one was already present (i.e. from a previous commit in the PR).

### CODE REVIEW

File: src/Commenter.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        ...context.repo,
+        issue_number,
+      })) {
+        // TODO: better type for comment
+        comment = comments.find((comment: any) => comment?.body?.includes(comment_tag_pattern));
+        if (comment) break;
+      }
+
+      if (comment) {
+        if (mode === MODE.Upsert) {
```
Comment (kgilpin) [inline src/Commenter.ts]: We don’t need a Mode option. Just update the comment each time :-)

**Comment by kgilpin** [comment]
@ahtrotta try this branch instead - https://github.com/land-of-apps/rails_tutorial_sample_app_7th_ed/pull/18

File: src/Commenter.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { GetResponseDataTypeFromEndpointMethod } from '@octokit/types';
+
+import assert from 'assert';
+import fs from 'fs';
+
+export default class Commenter {
+  private readonly githubToken: string;
+
+  constructor(private readonly filePath: string) {
+    this.githubToken = core.getInput('github-token');
```
Comment (kgilpin) [inline src/Commenter.ts]: `github-token` is defined here - https://github.com/getappmap/analyze-action/blob/0e8592c6f87c92792fc5250bd539163efcfd4c84/src/preflight.ts#L30 - so pass it in as a parameter. This makes it easier to test as well, and removes a dependency of this on `@actions/core`.

Comment (ahtrotta) [reply src/Commenter.ts]: Okay, that makes sense, I changed it.

**Pull request merged**